### PR TITLE
Build docker image in CI

### PR DESCRIPTION
**Pull request by julianoes**
From my limited bench testing, GCC version 10.3 seems to work, and saves some flash.

Next step is to update the docker containers, of course.

### CODE REVIEW

**Comment by davids5** [comment]
@julianoes - Do you think we should probably get some flight hours logged on this before the cut over.

**Comment by julianoes** [comment]
@davids5 I'd say yes, but also we're in-between release and many things are in flux, so it could just be part of that.

**Comment by julianoes** [comment]
Built on top of #19905

File: Tools/docker/Dockerfile-jammy [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ENV DEBIAN_FRONTEND noninteractive
+ENV LANG C.UTF-8
+ENV LC_ALL C.UTF-8
+
+RUN apt-get update && apt-get -y --quiet --no-install-recommends install \
+    ca-certificates \
+    wget
+
+ARG git_ref=main
+RUN wget https://raw.githubusercontent.com/PX4/PX4-Autopilot/$git_ref/Tools/setup/ubuntu.sh
```
Comment (dagar) [inline Tools/docker/Dockerfile-jammy:17]: Just copy this in? One of the motivations for moving this in tree was to keep all of this versioned together. It should be totally valid to jump back to an old branch and rebuild this (for whatever weird reason).

Comment (julianoes) [reply Tools/docker/Dockerfile-jammy:17]: Ah of course copy would be simpler! 🤦

**Comment by julianoes** [comment]
@dagar should we revive this or do we want it done differently?

**Pull request closed without merging**